components/camel-pulsar/src/test/java/org/apache/camel/component/pulsar/PulsarConcurrentProducerInTest.java

```diff
@@ -0,0 +1,119 @@
+ */
+package org.apache.camel.component.pulsar;
+
+import org.apache.camel.Endpoint;
+import org.apache.camel.EndpointInject;
+import org.apache.camel.Produce;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.component.pulsar.utils.AutoConfiguration;
+import org.apache.camel.impl.JndiRegistry;
+import org.apache.camel.test.junit4.CamelTestSupport;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.impl.ClientBuilderImpl;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+@Ignore //TODO use TestContainers to spin up local pulsar broker
+public class PulsarConcurrentProducerInTest extends CamelTestSupport {
+    private static final String PULSAR_CLUSTER_URL = "pulsar://localhost:6650";
+
+    private static final String TOPIC_URI = "persistent://public/default/camel-concurrent-producers-topic";
+    private static final String PRODUCER = "camel-producer";
+
+    @Produce(uri = "direct:start")
+    private ProducerTemplate producerTemplate;
+
+    @EndpointInject(uri = "pulsar:" + TOPIC_URI
+        + "?numberOfConsumers=3&subscriptionType=Shared"
+        + "&subscriptionName=camel-subscription&consumerQueueSize=1"
+        + "&consumerNamePrefix=camel-consumer"
+        + "&producerName=" + PRODUCER
+    )
+    private Endpoint from;
+
+    @EndpointInject(uri = "mock:result")
+    private MockEndpoint to;
+
+    @Override
+    protected RouteBuilder createRouteBuilder() {
+        return new RouteBuilder() {
+
+            @Override
+            public void configure() {
+                from("direct:start").to(from);
+                from(from).to(to);
+            }
+        };
+    }
+
+    @Override
+    protected JndiRegistry createRegistry() throws Exception {
+        JndiRegistry jndi = super.createRegistry();
+
+        registerPulsarBeans(jndi);
+
+        return jndi;
+    }
+
+    private void registerPulsarBeans(final JndiRegistry jndi) throws PulsarClientException {
+        PulsarClient pulsarClient = givenPulsarClient();
+        AutoConfiguration autoConfiguration = new AutoConfiguration(null, null);
+
+        jndi.bind("pulsarClient", pulsarClient);
+        jndi.bind("pulsar", new PulsarComponent(context(), autoConfiguration, pulsarClient));
+    }
+
+    private PulsarClient givenPulsarClient() throws PulsarClientException {
+        return new ClientBuilderImpl()
+            .serviceUrl(PULSAR_CLUSTER_URL)
+            .ioThreads(1)
+            .listenerThreads(1)
+            .build();
+    }
+
+    @Test
+    public void givenARunningPulsarCluster_whenIPublishAMessagesConcurrentlyToRoute_verifyMessageIsSentToClusterAndThenConsumed() throws Exception {
```

Comment:
i know naming is hard.  shorter name can be easy to read. like for example: testMultipleMessageConsumedByCluster

Reply:
agreed, modified these